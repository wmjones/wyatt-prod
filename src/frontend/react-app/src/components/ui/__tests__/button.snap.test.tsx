import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../button';

/**
 * Minimal button component snapshot tests
 *
 * These snapshots are kept minimal and focused on key visual states:
 * 1. Default button
 * 2. Primary variant (the most commonly used visual state)
 * 3. Disabled state (important user-visible state)
 *
 * We avoid creating excessive snapshots for every possible prop combination,
 * focusing only on states that are valuable to track for visual regression.
 */
describe('Button Component Snapshots', () => {
  it('renders default button correctly', () => {
    const tree = renderer
      .create(<Button>Default Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders primary variant button correctly', () => {
    const tree = renderer
      .create(<Button variant="primary">Primary Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot('primary button');
  });

  it('renders disabled button correctly', () => {
    const tree = renderer
      .create(<Button disabled>Disabled Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
