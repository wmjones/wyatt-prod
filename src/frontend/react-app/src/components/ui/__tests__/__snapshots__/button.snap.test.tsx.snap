// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots renders default button correctly 1`] = `
<button
  className="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
>
  Default Button
</button>
`;

exports[`Button Component Snapshots renders disabled button correctly 1`] = `
<button
  className="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
  disabled={true}
>
  Disabled Button
</button>
`;

exports[`Button Component Snapshots renders primary variant button correctly: primary button 1`] = `
<button
  className="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2"
>
  Primary Button
</button>
`;
